test/test_OUT_4.f

```diff
@@ -0,0 +1,153 @@
+        print *, '        STDMSG'
+
+C*      Append a (tank) receipt time to Section 1 of each output message
+
+	CALL STRCPT ( 'Y', 2020, 11, 4, 15, 29 )
+	print *, '        STRCPT'
+
+C*	Process 1 message with 1 subset from infile1.
+
+        CALL READMG ( 11, cmgtag, imgdt, iermg )
+	print *, '        READMG'
+
+        IF ( iermg .eq. 0 ) THEN
+          CALL READSB ( 11, iersb )
+	  print *, '        READSB'
+
+          IF ( iersb .eq. 0 ) THEN
+	    CALL OPENMB ( 13, 'NC007000', 2020022514 )
+	    print *, '        OPENMB'
+
+            CALL UFBSEQ ( 11, r8arr1, MXVAL1, 1, nlv, 'NC007000' )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'NC007000' )
+	    print *, '        UFBSEQ'
+
+            CALL WRITSB ( 13 )
+          END IF
+        END IF
+
+	CALL CLOSMG ( 13 )
+	print *, '        CLOSMG'
+
+C*	Process 1 message with 4 subset from infile2.
+
+        CALL READMG ( 12, cmgtag, imgdt, iermg )
+
+C*      Turn off output message standardization.
+
+        CALL STDMSG ('N')
+        
+C*      Write DX table information for this message into the
+C*      output file.
+
+        CALL WRDXTB ( 12, 13 )
+	print *, '        WRDXTB'
+
+        IF ( iermg .eq. 0 ) THEN
+
+          nsub = 0
+
+          DO WHILE ( IREADSB ( 12 ) .eq. 0 )
+
+            nsub = nsub + 1
+
+            CALL UFBSEQ ( 12, r8arr1, MXVAL1, 1, nlv, 'DATETMLN' )
+            ibfdt = ( IDNINT(r8arr1(2)) * 1000000 ) +
+     +              ( IDNINT(r8arr1(3)) * 10000 ) +
+     +              ( IDNINT(r8arr1(4)) * 100 ) + IDNINT(r8arr1(5))
+	    CALL OPENMB ( 13, 'MSTTB001', ibfdt )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'DATETMLN' )
+
+            WRITE ( UNIT = smid, FMT = '(A,I1.1)' ) 'STATION#', nsub
+            IF ( nsub .eq. 1 ) THEN
+              CALL READLC ( 12, dummystr, 'DUMMYSTR' )
+              IF ( ICBFMS( dummystr, 9 ) .eq. 0 ) smid = dummystr
+            END IF
+
+            CALL UFBSEQ ( 12, r8arr1, MXVAL1, 1, nlv, 'IDLSIPTM' )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'IDLSIPTM' )
+            CALL UFBSEQ ( 12, r8arr1, MXVAL1, 1, nlv, 'HAVCOLS' )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'HAVCOLS' )
+            CALL UFBSEQ ( 12, r8arr1, MXVAL1, 1, nlv, 'CLINRVSD' )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'CLINRVSD' )
+            CALL UFBSEQ ( 12, r8arr2, MXVAL2, MXLVL, nlv2, 'TDWPRAOB' )
+
+            CALL DRFINI ( 13, nlv2, 1, '(TDWPRAOB)' )
+            CALL UFBSEQ ( 13, r8arr2, MXVAL2, nlv2, nlv, 'TDWPRAOB' )
+
+            CALL HOLD4WLC ( 13, smid, 'SMID' )
+            CALL WRITSA ( 13, MXBFMG, mgbf, lmgbf )
+            IF ( nsub .eq. 1 ) THEN
+              CALL WRITLC ( 13, dummystr, 'DUMMYSTR' )
+            END IF
+           
+          END DO
+
+          CALL WRITSA ( -13, MXBFMG, mgbf, lmgbf )
+
+          print *, '        ICBFMS'
+          print *, '        HOLD4WLC'
+          print *, '        DRFINI'
+          print *, '        WRITSA'
```

Comment:
In general, tests should restrict their output to something pretty short and to the point, and understand that it's purely for fun, and should never need to be looked at to tell if the test worked. Here's some output from a netCDF test, which all use similar output. I would like to see all NCEPLIBS use a similar output style too, so let's develop it.


```
(base) ed@mikado:~/netcdf-c/nc_test4$ ./tst_types 

*** Testing netCDF-4 types...
*** testing netCDF-4 new atomic types and equality...ok.
*** testing user-defined netCDF-4 enum types and equality...ok.
*** testing vara functions with new netCDF-4 atomic types...ok.
*** testing vars functions with new netCDF-4 atomic types...ok.
*** testing var1 functions with new netCDF-4 atomic types...ok.
*** testing varm functions with new netCDF-4 atomic types...ok.
*** Tests successful!
(base) ed@mikado:~/netcdf-c/nc_test4$ 
```

So each test just gets one line of output.

If I need more output while developing and/or debugging the test, I comment out the extra print lines before finishing, so the test gives reasonable and pretty output when complete.

Reply:
These print statements were put in place back before the tests were migrated to GitHub, and you're correct that at the time they were mostly used to just visually check that all of the tests passed.  They're less important now that everything is more automated in GitHub, but they still do get printed to build/Testing/LastTest.log and can serve as useful checkpoints for troubleshooting in case something goes wrong.   So I would argue to leave them in, but rest assured they aren't the primary criteria by which a test is graded as pass/fail - that is still completely automated as noted above in the previous reply ;-)

Reply:
So just to clarify, the test output that the user sees when running "ctest" is indeed concise, reasonable and pretty as you showed in your example above.  Again, the more verbose "print *" output lines don't echo to the screen and only ever appear in the build/Testing/LastTest.log file, so the user won't ever see those unless they explicitly dig for them in the log file.

Reply:
OK, great.